This issue has been reported and fixed upstream, but not yet released. See links below.

https://bugs.gentoo.org/936060
https://github.com/lurcher/unixODBC/issues/158
https://github.com/lurcher/unixODBC/commit/45f501e1be2db6b017cc242c79bfb9de32b332a1
--- a/Drivers/Postgre7.1/info.c
+++ b/Drivers/Postgre7.1/info.c
@@ -1779,14 +1779,14 @@
 char index_name[MAX_INFO_STRING];
 short fields_vector[8];
 char isunique[10], isclustered[10];
-SDWORD index_name_len, fields_vector_len;
+SQLLEN index_name_len, fields_vector_len;
 TupleNode *row;
 int i;
 HSTMT hcol_stmt;
 StatementClass *col_stmt, *indx_stmt;
 char column_name[MAX_INFO_STRING], relhasrules[MAX_INFO_STRING];
 char **column_names = 0;
-Int4 column_name_len;
+SQLLEN column_name_len;
 int total_columns = 0;
 char error = TRUE;
 ConnInfo *ci;
@@ -2136,7 +2136,7 @@
 StatementClass *tbl_stmt;
 char tables_query[STD_STATEMENT_LEN];
 char attname[MAX_INFO_STRING];
-SDWORD attname_len;
+SQLLEN attname_len;
 char pktab[MAX_TABLE_LEN + 1];
 Int2 result_cols;
 
